### PR TITLE
Fix #976: Add support for websocket send_json and receive_json

### LINKED ISSUE #976: Fix websocket json support
Implement server `resp.send_json()` and client `resp.send_json()`/`resp.receive_json()`

### PR DESCRIPTION
**Pull request by vitaliemaldur**
## What do these changes do?

Add support for websocket server to send json and for websocket client to send and receive json

## Are there changes in behavior for the user?
User can use methods:
`send_json` and `receive_json` both on websocket server and client

## Related issue number
#976 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes

### CODE REVIEW

File: aiohttp/web_ws.py [hunk 1]
```diff
@@ -154,6 +154,13 @@ def send_bytes(self, data):
                             type(data))
         self._writer.send(data, binary=True)
 
+    def send_json(self, data, *, dumps=json.dumps):
```
Comment (asvetlov) [inline aiohttp/web_ws.py:157]: Please call `.send_str()` instead of working on low level.

File: aiohttp/websocket_client.py [hunk 2]
```diff
@@ -88,6 +89,11 @@ def send_bytes(self, data):
                             type(data))
         self._writer.send(data, binary=True)
 
+    def send_json(self, data, *, dumps=json.dumps):
+        if self._closed:
```
Comment (asvetlov) [inline aiohttp/websocket_client.py]: Use `send_str()`

**Comment by asvetlov** [comment]
Would you add tests also?

**Comment by vitaliemaldur** [comment]
Yes I will

**Comment by vitaliemaldur** [comment]
@asvetlov what I can do to avoid

```
AttributeError: 'module' object has no attribute 'JSONDecodeError'
tests\test_web_websocket.py:161: AttributeError
```

I am using `json.decoder.JSONDecodeError` in a `try/except`.

Also, do you have some suggestions for the tests I wrote?

File: aiohttp/web_ws.py [hunk 3]
```diff
@@ -154,6 +154,13 @@ def send_bytes(self, data):
...
+        if self._writer is None:
```
Comment (asvetlov) [inline aiohttp/web_ws.py]: These checks are not required: `.send_str()` performs them anyway.
There is no need to do the check twice.

File: tests/test_web_websocket.py [hunk 4]
```diff
@@ -128,6 +144,25 @@ def receive():
...
+    def test_receive_json_nonjson(self):
+
+        @asyncio.coroutine
+        def go():
+            req = self.make_request('GET', '/')
+            ws = WebSocketResponse()
+            yield from ws.prepare(req)
+
+            @asyncio.coroutine
+            def receive():
```
Comment (asvetlov) [inline tests/test_web_websocket.py]: Use functional test for the case.

**Comment by vitaliemaldur** [comment]
@asvetlov I've noticed that in `web_ws.py` we have `send_str`, `send_bytes`, `send_json` methods and coresponding `receive_str`, `receive_bytes`, `receive_json`. But in `websocket_client.py` we have `send_str`, `send_bytes`, `send_json` methods and only `receive_json`. Is that on purpose? or should I add the rest of methods there?

**Comment by asvetlov** [comment]
I think yes. 
From my perspective we should support the same methods for both client and server for consistency.

**Comment by vitaliemaldur** [comment]
@asvetlov I have added all the methods discussed above, now we have:
- `send_str`
- `send_bytes`
- `send_json`
- `receive_str`
- `receive_bytes`
- `receive_json`

both on `websocket_client.py` and `web_ws.py`. I have also added documentation for them.

**Comment by asvetlov** [comment]
Thanks!

**Pull request merged**